src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/Signatures.java

```diff
@@ -92,44 +92,61 @@ public static Content getPackageSignature(PackageElement pkg, PackageWriterImpl
     static class TypeSignature {
 
         private final TypeElement typeElement;
-        private final ClassWriterImpl classWriter;
+        private final HtmlDocletWriter writer;
         private final Utils utils;
         private final HtmlConfiguration configuration;
-        private Content modifiers;
+        private final Content modifiers;
 
-        TypeSignature(TypeElement typeElement, ClassWriterImpl classWriter) {
-            this.typeElement = typeElement;
-            this.classWriter = classWriter;
-            this.utils = classWriter.utils;
-            this.configuration = classWriter.configuration;
-        }
+        private static final Set<String> previewModifiers
+                = Set.of("sealed", "non-sealed");
```

Comment:
This needs to be coordinated with the work to make sealed final.
In the changes for that, this set is now empty but the code to process the set (lines 110-126) remains, ready for any future preview modifiers.

Also, given this code exists elsewhere, we should probably try and pull it in to a shared method.  That being said, Signatures is probably conceptually the right place for the shared method (not Utils.)

Also, whether or not we make a shared method, we should probably add an explanatory comment explaining why we keep an empty set around, just in case.

Reply:
> Generally: wow, very nice cleanup!
> 
> That being said, there is a potential conflict/race with a separate review to make sealed cases permanent (not preview.) So, this is approved, with a warning to coordinate with @vicente-romero-oracle about the sealed classes review #3613

Thanks for the review, Jon. @vicente-romero-oracle please go ahead with your PR, I'll update mine afterwards. (I should have been aware you were removing these sealed preview bits.)